src/gif.rs

```diff
@@ -93,13 +94,27 @@ impl<'a, R: 'a + Read> ImageDecoder<'a> for Decoder<R> {
     }
 
     fn read_image(mut self) -> ImageResult<Vec<u8>> {
-        if self.reader.next_frame_info()?.is_some() {
-            let mut buf = vec![0; self.reader.buffer_size()];
-            self.reader.read_into_buffer(&mut buf)?;
-            Ok(buf)
+        let (f_width, f_height, left, top);
+
+        if let Some(frame) = self.reader.next_frame_info()? {
+            left = u32::from(frame.left);
+            top = u32::from(frame.top);
+            f_width = u32::from(frame.width);
+            f_height = u32::from(frame.height);
         } else {
-            Err(ImageError::ImageEnd)
+            return Err(ImageError::ImageEnd);
         }
+
+        let mut buf = vec![0; self.reader.buffer_size()];
+        self.reader.read_into_buffer(&mut buf)?;
+
+        let image_buffer_raw = ImageBuffer::from_raw(f_width, f_height, buf).unwrap();
```

Comment:
`buffer_size` is input derived from the file. This `unwrap` should instead be a `match` or similar to check if the assumption was indeed correct.

Note that within `gif` the method `buffer_size()` involves converting the image dimensions (`u16`) to the native pointer size `usize` and multiplying several times. At 32-bit and 16-bit these are not necessarily lossless and also may not fit as asserted by `ImageBuffer::from` and thus crash here.

Reply:
I somehow assumed that `gif ` would reject image sizes that do not fit in `usize`. Will fix.